introspection/introspection.go

```diff
@@ -0,0 +1,289 @@
+package introspection
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/pprof"
+
+	"github.com/quay/clair/v4/config"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/key"
+	"go.opentelemetry.io/otel/exporter/metric/dogstatsd"
+	"go.opentelemetry.io/otel/exporter/metric/prometheus"
+	"go.opentelemetry.io/otel/exporter/trace/jaeger"
+	"go.opentelemetry.io/otel/exporter/trace/stdout"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+const (
+	Prom                  = "prometheus"
+	DefaultPromEndpoint   = "/metrics"
+	DogStatsD             = "dogstatsd"
+	STDOut                = "stdout"
+	Jaeger                = "jaeger"
+	DefaultJaegerEndpoint = "localhost:6831"
+	HealthEndpoint        = "/healthz"
+)
+
+// Introspection provides an http server
+// exposing Clair metrics and traces
+type Introspection struct {
+	// configuration provided when starting Clair
+	conf config.Config
```

Comment:
copied so that code cannot alter Config. 
IMO it's OK to optimize for writing less code here. these are init structures which are instantiated at most 1 time in our app. If we have memory pressure over copying the config twice, we have much larger problems.